cmd/blobovnicza-to-peapod/main.go

```diff
@@ -0,0 +1,95 @@
+	shardconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard"
+	blobovniczaconfig "github.com/nspcc-dev/neofs-node/cmd/neofs-node/config/engine/shard/blobstor/blobovnicza"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/blobovniczatree"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/common"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/compression"
+	"github.com/nspcc-dev/neofs-node/pkg/local_object_storage/blobstor/peapod"
+)
+
+func main() {
+	nodeCfgPath := flag.String("config", "", "Path to storage node's YAML configuration file")
+
+	flag.Parse()
+
+	if *nodeCfgPath == "" {
+		log.Fatal("missing storage node config flag")
+	}
+
+	appCfg := config.New(config.Prm{}, config.WithConfigFile(*nodeCfgPath))
+
+	err := engineconfig.IterateShards(appCfg, false, func(sc *shardconfig.Config) error {
+		log.Println("processing shard...")
+
+		var bbcz common.Storage
+		var perm fs.FileMode
+		storagesCfg := sc.BlobStor().Storages()
+
+		for i := range storagesCfg {
+			if storagesCfg[i].Type() == blobovniczatree.Type {
+				bbczCfg := blobovniczaconfig.From((*config.Config)(storagesCfg[i]))
+
+				perm = storagesCfg[i].Perm()
+				bbcz = blobovniczatree.NewBlobovniczaTree(
+					blobovniczatree.WithRootPath(storagesCfg[i].Path()),
+					blobovniczatree.WithPermissions(storagesCfg[i].Perm()),
+					blobovniczatree.WithBlobovniczaSize(bbczCfg.Size()),
+					blobovniczatree.WithBlobovniczaShallowDepth(bbczCfg.ShallowDepth()),
+					blobovniczatree.WithBlobovniczaShallowWidth(bbczCfg.ShallowWidth()),
+					blobovniczatree.WithOpenedCacheSize(bbczCfg.OpenedCacheSize()))
+
+				break
+			}
+		}
+
+		if bbcz == nil {
+			log.Println("Blobovnicza is not configured for the current shard, going to next one...")
+			return nil
+		}
+
+		bbczPath := bbcz.Path()
+		if !filepath.IsAbs(bbczPath) {
+			log.Fatalf("Blobobvnicza tree path '%s' is not absolute, make it like this in the config file first\n", bbczPath)
+		}
+
+		ppdPath := filepath.Join(filepath.Dir(bbcz.Path()), "peapod.db")
+		ppd := peapod.New(ppdPath, perm)
+
+		var compressCfg compression.Config
+		compressCfg.Enabled = sc.Compress()
+		compressCfg.UncompressableContentTypes = sc.UncompressableContentTypes()
+
+		err := compressCfg.Init()
+		if err != nil {
+			log.Fatal("init compression config for the current shard: ", err)
+		}
+
+		bbcz.SetCompressor(&compressCfg)
+		ppd.SetCompressor(&compressCfg)
+
+		log.Printf("migrating data from Blobovnicza tree '%s' to Peapod '%s'...\n", bbcz.Path(), ppd.Path())
+
+		err = common.Copy(ppd, bbcz)
+		if err != nil {
+			log.Fatal("migration failed: ", err)
+		}
+
+		log.Println("data successfully migrated in the current shard, going to the next one...")
+
+		return nil
+	})
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	log.Println("data successfully migrated in all shards, you may now re-configure node to work with Peapod")
```

Comment:
@roman-khimov, @cthulhu-rider, @532910 as i understood the [comment](https://github.com/nspcc-dev/neofs-node/pull/2462#discussion_r1291231421) suggested migrating config file too (remove useless lines, adjust types and so on). so we want it?

Reply:
Yeah, that'd be helpful in general and it's OK to do since config is explicitly given as a parameter.

Reply:
added, looks pretty messy, but should help